pkg/storageincentives/agent.go

```diff
@@ -323,12 +333,23 @@ func (a *Agent) play(ctx context.Context) (uint8, []byte, error) {
 
 	storageRadius := a.radius.StorageRadius()
 
+	// true if frozen
+	isFrozen, err := a.redistributionStatuser.IsOverlayFrozen(ctx)
+	if err != nil {
+		a.logger.Info("error checking if stake is frozen", "err", err)
+	}
+	a.nodeState.isFrozen(false)
```

Comment:
this should be in an else statement after line 342

Reply:
also a.nodeState.isFrozen should only be called when err is nil

Reply:
So by default, it should be `false`. `Else` statement is just an extra.

Reply:
If it's `true` then the old value will be overwritten.